test/compute_bands.jl

```diff
@@ -168,46 +162,27 @@ end
     ret = DFTK.prepare_band_data((basis=basis, λ=λ, λerror=λerror), klabels=klabels)
 
     @test ret.n_spin   == 1
-    @test ret.n_kcoord == 9
+    @test ret.n_kcoord == 8
     @test ret.n_bands  == 4
 
-    @test ret.branches[1].kindices == [1, 2, 3]
-    @test ret.branches[2].kindices == [4, 5, 6]
-    @test ret.branches[3].kindices == [7, 8, 9]
-
-    @test ret.branches[1].klabels == ("Γ", "X")
-    @test ret.branches[2].klabels == ("X", "W")
-    @test ret.branches[3].klabels == ("U", "X")
```

Comment:
In case we decide to keep the `kdistances_and_ticks` function, I'd still test the branch splitting explicitly by checking the branch index ranges in the `branches` field.

Reply:
Now `branches` is an input of `kdistances_and_ticks`.